ext/curl/interface.c

```diff
@@ -1634,7 +1634,16 @@ static bool php_curl_set_callable_handler(zend_fcall_info_cache *const handler_f
 #define HANDLE_CURL_OPTION_CALLABLE_PHP_CURL_USER(curl_ptr, constant_no_function, handler_type) \
 	case constant_no_function##FUNCTION: { \
 		bool result = php_curl_set_callable_handler(&curl_ptr->handlers.handler_type->fcc, zvalue, is_array_config, #constant_no_function "FUNCTION"); \
-		if (!result) { \
+		if (!result || (curl_ptr->handlers.handler_type && !curl_ptr->handlers.handler_type->fcc.function_handler)) { \
+			if (curl_ptr->handlers.read) { \
+                        	curl_ptr->handlers.read->method = PHP_CURL_STDOUT; \
+			} \
+			if (curl_ptr->handlers.write) { \
+                        	curl_ptr->handlers.write->method = PHP_CURL_DIRECT; \
+			} \
+			if (curl_ptr->handlers.write_header) { \
+                        	curl_ptr->handlers.write_header->method = PHP_CURL_IGNORE; \
+			} \
 			return FAILURE; \
```

Comment:
Setting the option to null shouldn't result in failure. Right now, you're also checking all three handlers for each call to `HANDLE_CURL_OPTION_CALLABLE_PHP_CURL_USER()`, which is called per handler.